labs/lab-2-javascript-game/script.js

```diff
@@ -0,0 +1,158 @@
+  playerHealth = 40;
+  enemyHealth = 10;
+  healsRemaining = 2;
+  winCount = 0;
+  roundCount = 1;
+}
+
+function roundReset(){
+  document.getElementById("playerStatusText").innerText = "";
+  document.getElementById("enemyStatusText").innerText = "";
+  enemy.health = intEnemyHealth;
+  calPlayerHealth();
+  calEnemyHealth();
+
+
+}
+
+function checkWin() {
+  if (winCount === 3) {
+    quitGame();
+    alert("You Won!")
+  } else if (player.health <= 0) {
+    quitGame();
+    alert("You lost!")
+  } else if (enemy.health === 0 && winCount < 3){
+    winCount++;
+    roundCount++;
+    roundReset();
+    return
+  }
+}
+
+function attack() {
+  var playerHealthPrevious = player.health;
+  var enemyHealthPrevious = enemy.health;
+  enemy.health = enemy.health - player.generateAttackDamage();
+  document.getElementById("playerStatusText").innerText =  "You dealt " + (enemyHealthPrevious - enemy.health) + " damage to " + enemy.name + ".";
+  calEnemyHealth();
+  player.health = player.health - enemy.generateAttackDamage();
+  document.getElementById("enemyStatusText").innerText =  enemy.name + " dealt " + (playerHealthPrevious - player.health) + " damage to " + player.name + ".";
+  calPlayerHealth();
+  checkWin();
+}
+
+function heal() {
+  if (player.healsRemaining > 0 && player.health !== 40){
+    var playerHealthPrevious = player.health;
+    player.health = player.health + player.heal();
+    document.getElementById("playerStatusText").innerText = "You healed "+ (player.health - playerHealthPrevious) + " health.";
+    player.healsRemaining--;
+    calPlayerHealth();
+    playerHealthPrevious = player.health;
+    player.health = player.health - enemy.generateAttackDamage();
+    document.getElementById("enemyStatusText").innerText =  enemy.name + " dealt " + (playerHealthPrevious - player.health) + " damage to " + player.name + ".";
+    calPlayerHealth();
+    checkWin();
+  } else if (player.health === 40) {
+    document.getElementById("statusMessage").innerText = "Your health is full."
+    return
+  } else if (player.healsRemaining === 0) {
+    document.getElementById("statusMessage").innerText = "You have no heals left."
+    return
+  }
+}
+
+function calPlayerHealth() {
+  if (player.health < 0) {
+    player.health = 0;
+  }
+  var playerHealthPercentage = (player.health / 40) * 100;
+  document.getElementById("playerHealthText").innerText = player.health+"/40";
+  document.getElementById("playerHealthBar").style.width = playerHealthPercentage+"%";
+  if (playerHealthPercentage > 75){
+    document.getElementById("playerHealthBar").style.backgroundColor = "green";
+  } else if (playerHealthPercentage < 75 && playerHealthPercentage > 40) {
+    document.getElementById("playerHealthBar").style.backgroundColor = "yellow";
+  } else if (playerHealthPercentage < 40) {
+    document.getElementById("playerHealthBar").style.backgroundColor = "red";
+  }
+}
+
+function calEnemyHealth() {
```

Comment:
A good chunk of `playerHealth` and `enemyHealth` could probably have been consolidated and added to the character class, if characters also stored their `maxHealth`.

I'd have pulled `healthColor(percentage)` into it's own function, too. Don't write more code than you gotta!